### ISSUE TITLE
Extend [platform.<target>.externals] overlay to Python, C#, Java, F# backends

### ISSUE THREAD

**Issue by Neftedollar** [Neftedollar/ll-lang#156]
## Context

Commit `fa25254` (closes #155) added per-project FFI overlay via \`lll.toml\`:

\`\`\`toml
[platform.typescript.externals]
\"expressApp\" = \"() => require('express')()\"
\"appGet\" = \"(app) => (path) => (handler) => { app.get(path, (req, res) => handler(req)(res)); return 0; }\"
\`\`\`

Only **TypeScript** codegen (\`CodegenTS.fs\`) currently splices overlay externals into output. Other targets (Python, C#, Java, F#, LLVM) don't yet support the overlay at emission time — validation falls back correctly but the generated code is incomplete.

## Scope

Extend the overlay to all target backends:

### Per-target
- [ ] \`CodegenPy.fs\` — emit \`name = <raw Python expr>\` for overlay externals; \`[platform.python.externals]\` in lll.toml
- [ ] \`CodegenCSharp.fs\` — emit \`public static dynamic <name> = <raw C# expr>;\` for overlay externals
- [ ] \`CodegenJava.fs\` — emit interface or static-method equivalent
- [ ] \`Codegen.fs\` (F#) — emit \`let <name> = <raw F# expr>\` for overlay externals
- [ ] \`CodegenLLVM.fs\` — emit \`declare <name>\` only; actual implementation provided by C runtime shim

### Demos that will benefit
- #152 ASP.NET — could move ASP.NET minimal API bindings (WebApplication.MapGet, etc.) to \`lll.toml\` and eliminate the Program.cs split
- #153 PyTorch — could declare torch.Tensor, torch.nn.Linear as overlays and keep all logic in \`.lll\`

### Acceptance
- \`examples/aspnet-todo\` rewritten to use \`[platform.csharp.externals]\` — no hand-written \`Program.cs\`
- \`examples/pytorch-mlp\` rewritten to use \`[platform.python.externals]\` — no hand-written \`train.py\` (or minimal stub)
- Regression: all 13 LLVM examples still pass; stdlib tests unaffected

### Non-goals
- Not changing the overlay syntax (TOML string values spliced raw)
- Not adding type-checking on overlay expressions (user's responsibility)

## Effort
Estimate: **4-6 hours per target** (5 targets × similar pattern to CodegenTS.fs changes).

## Labels
enhancement, ffi, codegen

**Comment by Neftedollar** [comment]
Implemented via the current FFI sidecar architecture (`FFI.lll`) rather than TOML-only overlay.

### What is now in place
- All major emitters resolve externals through `tryGetExternalTarget`:
  - `Codegen.fs` (F#)
  - `CodegenTS.fs` (TS)
  - `CodegenPy.fs` (Python)
  - `CodegenJava.fs` (Java)
  - `CodegenCSharp.fs` (C#)
  - `CodegenLLVM.fs` (LLVM declare path)
- Target FFI maps exist for each SDK:
  - `sdks/Platform.FSharp.SDK/src/FFI.lll`
  - `sdks/Platform.TypeScript.SDK/src/FFI.lll`
  - `sdks/Platform.Python.SDK/src/FFI.lll`
  - `sdks/Platform.Java.SDK/src/FFI.lll`
  - `sdks/Platform.CSharp.SDK/src/FFI.lll`
  - `sdks/Platform.LLVM.SDK/src/FFI.lll`

### Verification
- `dotnet test tests/LLLangTests/LLLangTests.fsproj --filter "FullyQualifiedName~PlatformSdkBuildTests"`
- Result: Passed 20, Failed 0
- Includes target-aware external mapping checks and FFI sidecar resolution/collision scenarios.

Given this, cross-target external emission is now covered; closing this issue in favor of the sidecar-based FFI flow.
